contracts/ContractRegistry.sol

```diff
@@ -1,22 +1,19 @@
-pragma solidity ^0.4.11;
+pragma solidity ^0.4.13;
 
-/*
- * @title Registry mapping keys (hash of contract name) with contract addresses
- */
-contract ContractRegistry {
-    mapping (bytes32 => address) registry;
+import "./Manager.sol";
 
-    function registrySet(bytes32 _key, address _contract) internal returns (bool) {
+import "zeppelin-solidity/contracts/lifecycle/Pausable.sol";
+
+contract ContractRegistry is Pausable {
+    mapping (bytes32 => address) public registry;
+
+    function setContract(bytes32 _key, address _contract) onlyOwner public returns (bool) {
         registry[_key] = _contract;
 
         return true;
     }
 
-    function registryGet(bytes32 _key) internal constant returns (address) {
-        return registry[_key];
-    }
-
-    function registryContains(bytes32 _key) internal constant returns (bool) {
-        return registry[_key] != address(0x0);
+    function updateManagerRegistry(bytes32 _key, address _registry) onlyOwner whenPaused public returns (bool) {
```

Comment:
Yeah all state mutating functions would need `whenNotPaused` modifiers to ensure that they can't be called during a pause. I wonder if there is a benefit in implementing those modifiers such that they will check if `LivepeerProtocol` (the main entry point into the system) is paused - if `LivepeerProtocol` is paused then all manager contracts are paused as well.

It might look like this:
```
modifier whenRegistryNotPaused() {
    require(!ContractRegistry(registry).paused);
    _;
}
```

This might make it easier to perform a system wide pause. Thoughts?